### ISSUE TITLE
Can't post due to codec error

### ISSUE THREAD

**Issue by M-Roy** [guoarthur/Kijiji-Repost-Headless#11]
I get a charmap codec error whenever I try to post an add.
I don't know if the fact that I'm trying to post in the Québec city section does that. I tried changing the font of the cmd panel, but that didn't fix it. 

![sans titre](https://cloud.githubusercontent.com/assets/28509180/25782386/2e5383d0-3318-11e7-87ec-4d9621bdd05e.png)

**Comment by StefGou** [comment]
Is your ad in french? It must be a problem with French accented characters encoding somewhere.

**Comment by M-Roy** [comment]
Yes the ad is in french. I didn't use any special characters though, since this was a test ad and I couldn't post in the past. I tried without using é, è, à or other special characters to try to make it work, but it didn't.

That's why I suspect Québec city category to be the problem.

Also, would it be possible to send an ad description using special characters (à, é, è) in this implementation ?

Thanks in advance and sorry if it's an easy fix, I'm really new to python !

**Comment by guoarthur** [comment]
Can you include the myAd.inf file you used to post this ad please?

**Comment by StefGou** [comment]
Looking at the KijijiApiException class, I think the problem might be that the "/tmp/kijiji-api-dump" folder doesn't exist?

Maybe modify the code to check if folder exists, else create it?

**Comment by M-Roy** [comment]
postAdForm.geocodeLat=46.7782027
postAdForm.geocodeLng=-71.3704154
postAdForm.city=Ville de Qu�bec
postAdForm.province=QC
PostalLat=46.7782027
PostalLng=-71.3704154
categoryId=214
postAdForm.adType=OFFER
postAdForm.priceType=FIXED
postAdForm.priceAmount=900
postAdForm.attributeMap[numberbathrooms_s]=10
postAdForm.attributeMap[forsaleby_s]=ownr
postAdForm.title=Grand 4 style condo a louer secteur cinema
postAdForm.description=Grand 4 et demi a louer secteur l'anciennce Lorette, pres des cinema.\nbeaucoup de rangement, 2 stationnements, etc.\ncontacter dan 313 322 2234
postAdForm.locationId=1700124
locationLevel0=9001
postAdForm.postalCode=G2G 1C2
featuresForm.topAdDuration=7
submitType=saveAndCheckout
imageCsv=picture1.jpg,picture2.jpg

The problem def. comes from the city I think.

**Comment by M-Roy** [comment]
@StefGou I had to change the dumpfile location since I am on a Windows based machine and that the directory didn't exist. I decided to just put my dumpfile in a dump.txt in the same folder. On the screenshot you can see the dumpfile empty.

**Comment by guoarthur** [comment]
This program was made for posting physical items, I haven't tested it on rental properties.

The fix is to add the following lines to your .inf file (You may want to change the last 2 lines based on preferences)

```
postAdForm.attributeMap[forrentbyhousing_s]=ownr
postAdForm.addressCity=Ville de Quebec
postAdForm.addressPostalCode=G2G 1C2
postAdForm.attributeMap[furnished_s]=1
postAdForm.attributeMap[petsallowed_s]=0
```

@StefGou Do you want to attempt to fix this so that people can post rentals in the future?

**Comment by StefGou** [comment]
@guoarthur I'll try in the next few days.

**Comment by M-Roy** [comment]
That's what a pro does.

Tried to fix this forever and you just did it hahaha ! Thank you so much, that did fix it.

**Comment by M-Roy** [comment]
Well, having this ''codec'' issue open,

For special characters, how could we enconde them so they get send properly to kijiji ?

(é, à, è ?)

**Comment by guoarthur** [comment]
@StefGou Feel free to reach out if you need anything!

@M-Roy I'll think about this, but probably will find a library which supports the utf8 encoding.

**Comment by StefGou** [comment]
@guoarthur Yeah, thanks. I'll probably need your help. :)

**Comment by jshapkin** [comment]
Are you on windows?  If so, this is likely the problem.  I can use this on Ubuntu, but not windows for this exact reason, no matter what city I am posting in.  I have not looked extensively into how to resolve this yet, but I will let you know if I find an easy fix

**Comment by M-Roy** [comment]
Yes exactly. I am on windows. I did not try it on other OS, so I think I'll just boot up a VM to use that for now, thanks for your insight !!

**Comment by vipertemp** [comment]
This is just what I was looking for! However I have one small problem similar to the user above: I can't input the unicode symbol ► in any of my posts

I'm very new to Python itself, and tried for something like ~8 hours and was able to make python make a text UTF-8 file with the symbol and read it back, and I could even make "python generate_inf_file.py" generate the UTF-8 file with my symbol in it. However, for some reason there was a problem with reaing back the '\n' line breaks  and reading it back in either caused your code to throw an error 

  File "kijiji_cmd.py", line 151, in <module>
    main()
  File "kijiji_cmd.py", line 57, in main
    args.function(args)
  File "kijiji_cmd.py", line 140, in repost_folder
    repost_ad(args)
  File "kijiji_cmd.py", line 119, in repost_ad
    [key, val] = line.strip().rstrip("\n").split("=")
ValueError: not enough values to unpack (expected 2, got 1)

OR if I used UTF-8, your code would generate the .inf file nicely, but kijiji would reject its contents when I tried to upload it. 

Otherwise if I don't use the ► symbol, your code runs very well. I'm not sure my comment is descriptive enough to help fix MY problem, but if there's some way for you to implement unicode, that would make my day.

Thanks!
